From 3d8d6dea4b36f638f40e50bb53fc2ac3cf14402b Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:58 -0400
Subject: [tools] perf/test: Update "sample parsing" test for PERF_SAMPLE_TRANSACTION

Message-id: <1408462094-14194-250-git-send-email-jolsa@redhat.com>
Patchwork-id: 88231
O-Subject: [PATCH RHEL7.1 BZ1131394 249/325] perf test: Update "sample parsing" test for PERF_SAMPLE_TRANSACTION
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 091a4ef5a94d46d26a05f0c32d2f64800ed91306
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Nov 1 15:51:37 2013 +0200

description
===========
In fact the "sample parsing" test does not automatically check new
sample type bits - they must be added to the comparison logic.

Doing that shows that the test fails because the functions
perf_event__synthesize_sample() and perf_event__sample_event_size() have
not been updated with PERF_SAMPLE_TRANSACTION either.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/sample-parsing.c b/tools/perf/tests/sample-parsing.c
index 61c9da2..1b67720 100644
--- a/tools/perf/tests/sample-parsing.c
+++ b/tools/perf/tests/sample-parsing.c
@@ -121,6 +121,9 @@ static bool samples_same(const struct perf_sample *s1,
  if (type & PERF_SAMPLE_DATA_SRC)
   COMP(data_src);
 
+ if (type & PERF_SAMPLE_TRANSACTION)
+  COMP(transaction);
+
  return true;
 }
 
@@ -165,6 +168,7 @@ static int do_test(u64 sample_type, u64 sample_regs_user, u64 read_format)
   .cpu  = 110,
   .raw_size = sizeof(raw_data),
   .data_src = 111,
+  .transaction = 112,
   .raw_data = (void *)raw_data,
   .callchain = &callchain.callchain,
   .branch_stack = &branch_stack.branch_stack,
@@ -273,7 +277,8 @@ int test__sample_parsing(void)
 
  /*
   * Fail the test if it has not been updated when new sample format bits
-  * were added.
+  * were added.  Please actually update the test rather than just change
+  * the condition below.
   */
  if (PERF_SAMPLE_MAX > PERF_SAMPLE_TRANSACTION << 1) {
   pr_debug("sample format has changed, some new PERF_SAMPLE_ bit was introduced - test needs updating\n");
-- 
1.7.1